airsonic-main/src/main/java/org/airsonic/player/service/AudioScrobblerService.java

```diff
@@ -285,39 +381,66 @@ private RegistrationThread() {
         public void run() {
             while (true) {
                 RegistrationData registrationData = null;
+                int succeedCount = 0;
                 try {
                     registrationData = queue.take();
-                    scrobble(registrationData);
+                } catch (Exception x) {
+                    LOG.error("Failed to deque registration.", x);
+                }
+
+                LOG.info("Begin registration for " + registrationData.title + ".");
+
+                try {
+                    scrobbleLastFm(registrationData);
+                    ++succeedCount;
+                    LOG.info("Succeed in Last.fm registration for " + registrationData.title + ".");
                 } catch (IOException x) {
-                    handleNetworkError(registrationData, x);
+                    LOG.warn("Network error in Last.fm registration for " + registrationData.title + ".", x);
```

Comment:
The logging in these try..catch blocks probably doesn't need a full stack trace of the exception, which is what you get when you pass the exception to the log message.  I would suggest using `x.toString()` instead to get just the error details.